whalesong/driver.py

```diff
@@ -141,8 +142,18 @@ def process_result_sync(self, result):
             self.logger.exception(ex)
 
     async def close(self):
-        await self._internal_close()
-        self._fut_stop.set_result(None)
+        if self._fut_start is not None:
+            await self._fut_start
+        else:
+            return  # Not started
+
+        if self._fut_stop:
+            await self._fut_stop
+            return
+
+        self._fut_start = None
```

Comment:
It must be last statement in order to avoid to restart driver before stop it.

Reply:
Tried your suggestion . It checks out . Plus ensure_future causes driver to close faster .